### PR TITLE
Implement nrm filename toml input, as well as renaming and trace mode configs

### PR DESCRIPTION
**Pull request by LittleCube-hax**
Might suck but maybe it's cool?

### CODE REVIEW

**Comment by LittleCube-hax** [comment]
i threw in renaming and RE

File: src/elf.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ bool read_symbols(N64Recomp::Context& context, const ELFIO::elfio& elf_file, ELF
 
         if (section_index < context.sections.size()) {        
             // Check if this symbol is the entrypoint
-            if (elf_config.has_entrypoint && value == elf_config.entrypoint_address && type == ELFIO::STT_FUNC) {
+            if (elf_config.has_entrypoint && (uint32_t) value == (uint32_t) elf_config.entrypoint_address && (type == ELFIO::STT_FUNC || type == ELFIO::STT_NOTYPE)) {
```
Comment (Mr-Wiseguy) [inline src/elf.cpp:63]: Why the changes here and below to include NOTYPE?

File: src/main.cpp [hunk 2]
```diff
@@ -273,10 +273,10 @@ int main(int argc, char** argv) {
     };
 
     // TODO expose a way to dump the context from the command line.
-    bool dumping_context = false;
```
Comment (Mr-Wiseguy) [inline src/main.cpp:276]: I'd prefer if this looked for an explicit flag, like `--dump-symbols`. The argc count change can stay, but the additional arguments should be checked against a flag list (which should just be this single flag for now).

**Review by Mr-Wiseguy — CHANGES_REQUESTED** [review]
Great work on the renaming functionality and the mod toml changes, and RE mode is really interesting as well. I do think the name should be tweaked, since RE mode is a bit unclear (at first I thought it was related to the renaming functionality). A more descriptive name would be better to illustrate exactly what it does. Maybe "function_trace_mode"?

**Review by Mr-Wiseguy — APPROVED** [review]

**Pull request merged**